kore/src/main/scala/org/kframework/definition/outer-ext.scala

```diff
@@ -56,12 +56,14 @@ object FlatModule {
       }
       memoization.getOrElseUpdate(m.name, {
         // transform all imports into Module
-        val newImports = m.imports.map(i => memoization.getOrElse(i.name
+        val f = (i: Import) => memoization.getOrElse(i.name
           // if can't find the Module in memoization, build a new one
           , toModuleRec(allModules.find(f => f.name.equals(i.name))
               .getOrElse(throw KEMException.compilerError("Could not find module: " + i.name, i))
-            , visitedModules :+ m)))
-        val newM = new Module(m.name, newImports, m.localSentences, m.att)
+            , visitedModules :+ m))
+        val newPublicImports = m.imports.filter(_.isPublic).map(f)
+        val newPrivateImports = m.imports.filter(!_.isPublic).map(f)
+        val newM = new Module(m.name, newPublicImports, newPrivateImports, m.localSentences, m.att)
```

Comment:
This construction pattern (split into public/private sentences, then build new Module type) seems common. Can we extract a "smart constructor" for `Module` which encapsulates that logic (so you just pass in the full set of sentences, and it does the split and constructs the module with the split sentences)?

Reply:
In point of fact, while there are three places in this PR that do something similar like this, they are enough different that they would not benefit from refactoring out a common function.